compiler/ilgen/TypeDictionary.hpp

```diff
@@ -229,6 +229,176 @@ class TypeDictionary
+    */
+   template <typename T>
+   struct is_supported {
+      static const bool value =  std::is_arithmetic<T>::value // note: is_arithmetic = is_integral || is_floating_point
+                              || std::is_void<T>::value;
+   };
+   template <typename T>
+   struct is_supported<T*> {
+      // a pointer type is supported iff the type being pointed to is supported
+      static const bool value =  is_supported<T>::value;
+   };
+
+   /** @fn template <typename T> TR::IlType* toIlType()
+    * @brief Given a C/C++ type, returns a corresponding TR::IlType, if available
+    * @tparam C/C++ type
+    * @return TR::IlType instance corresponding to the specified C/C++ type
+    * 
+    * Given a C/C++ type, `toIlType<>` will attempt to match the type with a
+    * corresponding TR::IlType instance that is usable with JitBuilder. If no
+    * type is **known** to match the specified type (meaning the type is
+    * unsupported), then the function call fails to compile.
+    * 
+    * Currently, only the following types are supported:
+    * 
+    * - all integral types (int, long, etc.)
+    * - all floating point types (float, double)
+    * - void
+    * - pointers to all the above types
+    * 
+    * Note that many user defined types (e.g. structs and arrays) are not
+    * currently supported.
+    * 
+    * For convenience, the template class `is_supported` can be used to determine
+    * at compile time whether a particular type is supported.
+    * 
+    * ## Usage
+    * 
+    * Using this template function is as simple as:
+    * 
+    * ```c++
+    * auto d = TR::TypeDictionary{};
+    * auto t = d.toIlType<int>();
+    * ```
+    * 
+    * If the type specified has no known corresponding TR::IlType, then the function
+    * call simply fails to compile, reporting that there is "no matching function
+    * call" (or some variation thereof).
+    * 
+    * ## Design
+    * 
+    * `toIlType<>()` is an overloaded template function. It uses SFINAE and the C++11
+    * type traits library to enable a specific overload (function implementation)
+    * that will return an instance of TR::IlType. Type traits are used to define
+    * rules that a type must follow to match a given function implementation.
+    * 
+    * For integer and floating point types, the type traits library is used to
+    * identify the "family" of the specified type. For example, `std::is_integral<>`
+    * is used to identify all the integer types. The `sizeof` operator is then used
+    * to determine the size of the specified type. The combination of the type's
+    * family and size is used to select the function that gets selected.
+    * 
+    * For types that do not belong to a family (e.g. `void`), the size is not needed.
+    * 
+    * For pointer types, `std::remove_pointer<>` is used to get the type being
+    * pointed to. Iff `is_supported<>::value` evaluates to true for this type,
```

Comment:
Typo: "Iff", unless you meant "If and only if" in which case the sentence is slightly confusing.